test/src/test/java/org/corfudb/integration/LogReplicationListenerIT.java

```diff
@@ -0,0 +1,667 @@
+                List<CorfuStreamEntry> entries = updatesPerTable.getOrDefault(entry.getKey().getTableName(),
+                    new ArrayList<>());
+                entries.addAll(entry.getValue());
+                updatesPerTable.put(entry.getKey().getTableName(), entries);
+            }
+        }
+
+        // Verify that the streaming data received for each table with the corresponding original table
+        for (String tableName : updatesPerTable.keySet()) {
+            log.info("TableName {}", tableName);
+            // Verify the actual data
+            for (CorfuStreamEntry entry : updatesPerTable.get(tableName)) {
+                    SampleSchema.Uuid key = (SampleSchema.Uuid) entry.getKey();
+                    SampleSchema.SampleTableAMsg val = (SampleSchema.SampleTableAMsg) entry.getPayload();
+                    try (TxnContext txn = store.txn(namespace)) {
+                        Assert.assertEquals(txn.getRecord(tableName, key).getPayload(), val);
+                        txn.commit();
+                    }
+
+            }
+        }
+
+        for (CorfuStoreEntry entry : existingEntries) {
+            try (TxnContext txn = store.txn(namespace)) {
+                SampleSchema.Uuid key = (SampleSchema.Uuid) entry.getKey();
+                SampleSchema.SampleTableAMsg val = (SampleSchema.SampleTableAMsg) entry.getPayload();
+                Assert.assertEquals(txn.getRecord(userTableName, key).getPayload(), val);
+            }
+        }
+    }
+
+    @After
+    @Override
+    public void cleanUp() throws Exception {
+        if (lrListener != null) {
+            store.unsubscribeListener(lrListener);
+        }
+        if (listener != null) {
+            store.unsubscribeListener(listener);
+        }
+        super.cleanUp();
+    }
+
+    private class TestListener implements StreamListener {
+        private CountDownLatch countDownLatch;
+
+        @Getter
+        private final LinkedList<CorfuStreamEntries> updates = new LinkedList<>();
+
+        TestListener(CountDownLatch countDownLatch) {
+            this.countDownLatch = countDownLatch;
+        }
+
+        @Override
+        public void onNext(CorfuStreamEntries results) {
+            results.getEntries().forEach((key, val) -> countDownLatch.countDown());
+            updates.add(results);
+        }
+
+        @Override
+        public void onError(Throwable throwable) {
+            log.error("Error on listener", throwable);
+            fail("onError for CrossNamespaceListener: " + throwable.toString());
+        }
+    }
+
+    private class LRTestListener extends LogReplicationListener {
+        private CountDownLatch countDownLatch;
+
+        // Updates received through streaming
+        @Getter
+        private final LinkedList<CorfuStreamEntries> updates = new LinkedList<>();
+
+        // Entries discovered in mergeTableOnSubscription() before streaming updates are received
+        @Getter
```

Comment:
nit: `mergeTableOnSubscription()` -> `performFullSync()`? same for L120 I thought